pkg/probe/turbo_probe.go

```diff
@@ -70,38 +73,32 @@ func newTurboProbe(probeConf *ProbeConfig) (*TurboProbe, error) {
 
 	myProbe := &TurboProbe{
 		ProbeConfiguration: probeConf,
-		DiscoveryClientMap: make(map[string]*TargetDiscoveryAgent),
+		DiscoveryClient:    nil, // DiscoveryClient is originally empty. Should be initialized later.
```

Comment:
nit: would it be better to initialize the `DiscoveryClient` here, since it is mandatory anyway?

```
DiscoveryClient: NewDiscoveryAgent(),
```